src/main/kotlin/uk/gov/dluhc/printapi/jobs/BatchPrintRequestsJob.kt

```diff
@@ -1,17 +1,24 @@
 package uk.gov.dluhc.printapi.jobs
 
 import mu.KotlinLogging
+import net.javacrumbs.shedlock.spring.annotation.SchedulerLock
+import org.springframework.beans.factory.annotation.Value
 import org.springframework.scheduling.annotation.Scheduled
 import org.springframework.stereotype.Component
+import uk.gov.dluhc.printapi.service.PrintRequestsService
 
 private val logger = KotlinLogging.logger {}
 
 @Component
-class BatchPrintRequestsJob {
+class BatchPrintRequestsJob(
+    private val printRequestsService: PrintRequestsService,
+    @Value("\${jobs.batch-print-requests.batch-size}")
+    private val batchSize: Int
+) {
 
-    @Scheduled(cron = "\${jobs.print-requests.cron}")
-//    @SchedulerLock(name = "\${jobs.print-requests.name}", lockAtMostFor = "\${jobs.print-requests.lockAtMostFor}", lockAtLeastFor = "\${jobs.print-requests.lockAtLeastFor}")
+    @Scheduled(cron = "\${jobs.batch-print-requests.cron}")
+    @SchedulerLock(name = "\${jobs.batch-print-requests.name}")
     fun run() {
-        logger.info { "Hello, I'm your batching agent" }
+        printRequestsService.processPrintRequests(batchSize)
```

Comment:
Is there any `Integration test` that actually calls this endpoint and asserts that print requests are processed ?

Reply:
There's no integration test for this yet ....
As you know I favour incremental PRs - Rome wasn't built in a day and all that :D - We need to get this PR merged for several reasons. We'll build upon it and add integration tests as we go 👍